File: games/story/adapters/csharp/AaaSystems.cs
namespace BizX.Game;
public sealed class AaaSystems { public int Tick {get;private set;}=0; public Dictionary<string,double> Prices {get;}=new(); public List<Npc> Npcs {get;}=new(); public sealed class Npc { public string Id=""; public string Faction=""; public double Food=1,Rest=1,Safety=1; public string State="idle"; }
 public void TickOnce(){Tick++;foreach(var n in Npcs){n.Food=Math.Max(0,n.Food-.005);n.Rest=Math.Max(0,n.Rest-.003);n.State=n.Safety<.25?"flee":n.Food<.25?"seek_food":n.Rest<.2?"sleep":"work";}}
 public double Market(string good,double supply,double demand){var p=Math.Max(.01,(demand+1)/(supply+1));Prices[good]=p;return p;}
 public int ResolveAbility(int baseDamage,int defense,int roll)=>roll+5<defense?0:baseDamage*(roll==20?2:1);
}

File: games/story/adapters/ruby/aaa_systems.rb
module BizX; module Game
  class AaaSystems
    attr_reader :tick,:npcs,:prices
    def initialize; @tick=0; @npcs=[]; @prices={}; end
    def tick_once; @tick+=1; @npcs.each{|n| n[:food]=[0,n[:food]-.005].max;n[:rest]=[0,n[:rest]-.003].max;n[:state]=n[:safety]<.25 ? 'flee' : n[:food]<.25 ? 'seek_food' : n[:rest]<.2 ? 'sleep' : 'work'}; end
    def market(good,supply,demand); @prices[good]=[.01,(demand+1.0)/(supply+1.0)].max; end
    def resolve_ability(base,defense,roll); roll+5<defense ? 0 : base*(roll==20 ? 2 : 1); end
  end
end; end
